aeneas/src/x86/X86CodeGen.v3

```diff
@@ -636,13 +743,30 @@ class X86CodeGen extends MachCodeGen {
 		gen(lt.0, asm_loadx, (lt.1, dfngpr(dest), gpr(base), scale, disp, canTrap, source));
 	}
 	def gen_loada(dest: VReg, t: Type, addr: Addr) {
+		if (t.typeCon.kind == V3Kind.FLOAT) return gen_loada_sse(dest, t, addr);
 		var lt = matchLoad(t, mach.sizeOf(t));
 		gen(lt.0, asm_loada, (lt.1, dfngpr(dest), addr));
 	}
 	def gen_loada_disp(dest: VReg, t: Type, addr: Addr, disp: SsaInstr) {
+		if (t.typeCon.kind == V3Kind.FLOAT) return gen_loada_disp_sse(dest, t, addr, disp);
 		var lt = matchLoad(t, mach.sizeOf(t));
 		gen(lt.0, asm_loada_disp, (lt.1, dfngpr(dest), addr, gpr(disp)));
 	}
+	def gen_loadx_sse(dest: VReg, t: Type, base: SsaInstr, scale: byte, disp: int, canTrap: bool, source: Source) {
+		if (mach.sizeOf(t) == 4) {
```

Comment:
Can you factor out {mach.sizeOf(t) == 4} to an {isDouble()} routine and invert these cases?